### ISSUE TITLE
What does 'sample_boundary.append(neg_seed * alpha + pos_seed.unsqueeze(2) * (1-alpha))' mean

### ISSUE THREAD

**Issue by jsor2009** [vita-epfl/social-nce#8]
Thanks for your hard work.

I have some questions about this code
            for alpha in alpha_list:
                sample_boundary.append(neg_seed * alpha + pos_seed.unsqueeze(2) * (1-alpha))
Looks like
          α* neg_ seeds + (1- α)* pos_ seeds
What is the purpose of this code

from sampling.py

        # primary-neighbor boundary
        if self.num_boundary > 0:
            alpha_list = torch.linspace(self.ratio_boundary, 1.0, steps=self.num_boundary) #0.5-1 分成0-9份 参数决定
            sample_boundary = []
            for alpha in alpha_list:
                sample_boundary.append(neg_seed * alpha + pos_seed.unsqueeze(2) * (1-alpha))
            sample_boundary = torch.cat(sample_boundary, axis=2)
            sample_neg = torch.cat([sample_boundary, sample_territory], axis=2)
        else:
            sample_neg = sample_territory

Thanks!

**Comment by YuejiangLIU** [comment]
Thanks for your interest.

The goal of this block is to draw some additional **_hard negative_** samples around the boundary between the standard positive samples (ego-agent territory) and standard negative samples (i.e. neighbor territory).

It's a variant of the sampling strategy reported in our paper, not activated in our default args setting. 

https://github.com/vita-epfl/social-nce/blob/5232d4de81ae8eab3ff12e500c9d7ac1e524e874/crowd_nav/imitate.py#L30

**Comment by jsor2009** [comment]
Thank you for your answers and your excellent work
